Source/JavaScriptCore/runtime/SetPrototype.cpp

```diff
@@ -175,6 +181,175 @@ JSC_DEFINE_HOST_FUNCTION(setProtoFuncSize, (JSGlobalObject* globalObject, CallFr
     return JSValue::encode(jsNumber(set->size()));
 }
 
+// https://tc39.es/ecma262/#sec-getsetrecord ( Step 1 ~ Step 7 )
+static uint32_t getSetSizeAsInt(JSGlobalObject* globalObject, JSValue value)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    if (!value.isObject()) [[unlikely]]
+        return throwVMTypeError(globalObject, scope, "Set operation expects first argument to be an object");
+
+    JSObject* obj = asObject(value);
+
+    JSValue rawSize = obj->get(globalObject, vm.propertyNames->size);
+    RETURN_IF_EXCEPTION(scope, 0);
+
+    double numSize = rawSize.toNumber(globalObject);
+    RETURN_IF_EXCEPTION(scope, 0);
+
+    if (std::isnan(numSize)) [[unlikely]]
+        return throwVMTypeError(globalObject, scope, "Set operation expects first argument to have non-NaN 'size' property"_s);
+
+    double intOrInfSize = jsNumber(numSize).toIntegerOrInfinity(globalObject);
+
+    if (intOrInfSize < 0) [[unlikely]]
+        return throwVMRangeError(globalObject, scope, "Set operation expects first argument to have non-negative 'size' property"_s);
+
+    if (std::isinf(intOrInfSize)) [[unlikely]]
+        return std::numeric_limits<uint32_t>::max();
+    return static_cast<uint32_t>(intOrInfSize);
+}
+
+static EncodedJSValue fastSetIntersection(JSGlobalObject* globalObject, JSSet* thisSet, JSSet* otherSet)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSSet* result = JSSet::create(vm, globalObject->setStructure());
+
+    JSSet* sourceSet = thisSet->size() <= otherSet->size() ? thisSet : otherSet;
+    JSSet* targetSet = thisSet->size() <= otherSet->size() ? otherSet : thisSet;
+
+    JSCell* sourceStorageCell = sourceSet->storageOrSentinel(vm);
+    if (sourceStorageCell == vm.orderedHashTableSentinel())
+        return JSValue::encode(result);
+
+    JSSet::Storage& sourceStorage = *jsCast<JSSet::Storage*>(sourceStorageCell);
+    JSSet::Helper::Entry entry = 0;
+
+    while (true) {
+        sourceStorageCell = JSSet::Helper::nextAndUpdateIterationEntry(vm, sourceStorage, entry);
+        if (sourceStorageCell == vm.orderedHashTableSentinel())
+            break;
+
+        JSSet::Storage& currentStorage = *jsCast<JSSet::Storage*>(sourceStorageCell);
+        entry = JSSet::Helper::iterationEntry(currentStorage) + 1;
+        JSValue entryKey = JSSet::Helper::getIterationEntryKey(currentStorage);
+
+        if (targetSet->has(globalObject, entryKey)) {
+            result->add(globalObject, entryKey);
+            RETURN_IF_EXCEPTION(scope, JSValue::encode(result));
+        }
+
+        sourceStorage = currentStorage;
+    }
+    return JSValue::encode(result);
+}
+
+JSC_DEFINE_HOST_FUNCTION(setProtoFuncIntersection, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSSet* thisSet = getSet(globalObject, callFrame->thisValue());
+    RETURN_IF_EXCEPTION(scope, { });
```

Comment:
This is not necessary.

Reply:
if second arg isn't set object, `getSet` throws `TypeError` and returns `nullptr` so I think this is necessary. In fact, if `RETURN_IF_EXCEPTION` is removed, a segv will occur.

Reply:
test262 includes that case ( Of course, I can add such cases to stress tests )

Reply:
Oh, that's right.